app/src/main/res/xml/preferences.xml

```diff
@@ -29,12 +29,11 @@
             android:entryValues="@array/number_decimals"/>
 
         <ListPreference
-            android:key="group_separator"
+            android:key="group_separator_v2"
             android:title="@string/prefs_title_group_separator"
             android:summary="@string/prefs_summary_group_separator"
-            android:defaultValue="@string/default_group_separator"
```

Comment:
Previously languages could set different defaults based on their own language conventions, and many took the chance to set a period (except for Japanese, which uses a comma). Now all languages are forced to use the default of no separator. In Hungarian, for example, numbers greater than 9999 must use a group separator of either period or a space; unseparated are spelling mistakes. You could make the _index_ localizable instead of the raw displayed string to still avoid the lost value issue. Probably a fifth option of “language default” could be added to the preference, which would be the default value, and would use this language-dependent default.

Reply:
That's a great idea, and something I'd not considered. Let me look into updating this, thanks for the suggestion.

Reply:
I wasn't able to get the default group separator by including it as a resource for all cases, ultimately due to the fact that I implemented `Preferences` as a singleton. I'm planning on refactoring that, but in the meantime I've implemented the default group separator as a property in the `Language` model which achieves the same effect.